### DISCUSSION TITLE
Deduplication?

### DISCUSSION THREAD

**Discussion by cowhead2000** [cloudamqp/lavinmq#439] [Ideas]
Are there any plans on deduplication functionality?

**Comment by carlhoerberg** [comment]
What are your requirements for msg dedup? Exchange level, queue level? How long back should it be able to dedup? Only body or headers too? All headers?

**Reply by cowhead2000** [reply]
Hi @carlhoerberg

Ideally, we should be able to set it up for both the exchange and the queue, but i'll settle for the exchange tho, if need be. A simple approach would be to use the header as key. Using same x-deduplication-headers (only dedupe these), x-cache-size (cache size in objects) with a optional x-cache-ttl (cache in milliseconds) as the deduplication plugin for rabbitmq.

Regards

**Reply by alistairstpierre** [reply]
Same request from me also.

Just need a simple queue level duplication check on the message contents.

**Reply by snichme** [reply]
Better late than never, here is a PR that introduce message deduplication: https://github.com/cloudamqp/lavinmq/pull/854 

It's under review but the interface (argument and header keys) will most likely stay the same, You are welcome to test it out.
